### DISCUSSION TITLE
How  to bind the EGLContext of the filament and the GLSurfaceView?

### DISCUSSION THREAD

**Discussion by dugufly** [google/filament#4643] [Q&A]
I want to bind the EGLContext of the filament and   GLSurfaceView, and then share the textures between filament and the GLSurfaceView.   Is there a corresponding sample code?

**Comment by romainguy** [comment]
You can pass the EGLContext you want to share when calling `Engine.create()`. Note however that this won't help you much as Filament doesn't expose its internal GL handles. You can however import GL textures `Texture.Builder.import()`. I would recommend using `SurfaceTexture` and external textures/`samplerExternal` instead though.

**Reply by dugufly** [reply]
Why the function "Texture.Builder.import()" subject to change or removal?

`

         * @warning This method should be used as a last resort. This API is subject to change or
         * removal.
         *
         * @param id a backend specific texture identifier
         * 
         *

         * @return This Builder, for chaining calls.
         * 
         */

        Builder& import(intptr_t id) noexcept;

`

**Reply by romainguy** [reply]
Because we may not support OpenGL forever.

**Reply by dugufly** [reply]
I use importTexture to import a texture（Create in GLSurfaceView） as the rendertarget. If the renderer.render (view) is completed, can the imported texture Id be used in GLSurfaceView? Is the data on the texture ID ready?

**Reply by romainguy** [reply]
It would be better to have Filament render into a `SurfaceTexture`'s `Surface` and use that surface texture as an external texture in your GL surface view. And you don't have to worry about synchronization in those cases with GL.
